### PR TITLE
Use NPP to convert from RGB to YCbCr in nvjpeg decoder

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes a the internal usage of nvjpeg for YCbCr outputs. Instead, we will decode to RGB and then convert to YCbCr with NPP.

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Start using NPP _Ctx APIs*
     *nvjpeg decoder: Decoding to RGB and using NPP in to convert from RGB to YCbCr*
 - Affected modules and functionalities:
     *ImageDecoder*
 - Key points relevant for the review:
     *Logic in nvjpeg decoder*
 - Validation and testing:
     *Exsiting tests*
 - Documentation (including examples):
     *NA*

**JIRA TASK**: *[DALI-1976]*

### CODE REVIEW

File: dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h [hunk 1]
```diff
@@ -770,6 +769,24 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
...
+        // what JPEG uses.
+        for (auto *sample : samples_hw_batched_) {
+          int i = sample->sample_idx;
+          auto data = output.mutable_tensor<uint8_t>(i);
+          auto sh = output_shape_.tensor_shape(i);
+          NppiSize size;
+          size.height = sh[0];
+          size.width = sh[1];
+          int step = sh[1] * 3;
+          DALI_CHECK_NPP(nppiRGBToYCbCr_8u_C3R_Ctx(data, step, data, step, size,
```
Comment (JanuszL) [inline dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h:785]: Does it work inplace?

Comment (JanuszL) [inline dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h:785]: Maybe you can extract this to a function and call in L921 as well.

File: dali/operators/image/color/color_space_conversion.cu [hunk 2]
```diff
@@ -164,41 +161,41 @@ void ColorSpaceConversion<GPUBackend>::RunImpl(DeviceWorkspace &ws) {
       } else if (conversion == kRGB_TO_YCbCr) {
         // RGB -> YCbCr
         DALI_CHECK_NPP(
-          nppiRGBToYCbCr_8u_C3R(
-            input_data, nStepInput, output_data, nStepOutput, size));
+          nppiRGBToYCbCr_8u_C3R_Ctx(
```
Comment (JanuszL) [inline dali/operators/image/color/color_space_conversion.cu:164]: 👍

File: dali/util/npp.cc [hunk 3]
```diff
@@ -35,4 +35,40 @@ int NPPInterpForDALIInterp(DALIInterpType type, NppiInterpolationMode *npp_type)
   return DALISuccess;
 }
 
+namespace {
+
+NppStreamContext GetNppContextImpl() {
+  NppStreamContext ctx;
+  ctx.hStream = nullptr;
```
Comment (JanuszL) [inline dali/util/npp.cc:42]:
```suggestion
  ctx.hStream = 0;
```
It is a pointer internally but 0 is always referenced as a default one.

**Review by JanuszL — APPROVED** [review]

**Review by JanuszL — APPROVED** [review]

**Comment by jantonguirao** [comment]
Closing in favor of 2907

**Pull request closed without merging**